packages/runner/src/lib/util.js

```diff
@@ -19,6 +15,10 @@ export default {
     return ''
   },
 
+  relativeSpecPath (config) {
+    return path.relative(config.fileServerFolder, this.absoluteSpecPath(config))
```

Comment:
The problem with this method is that it is using `this.absoluteSpecPath()` which the implementation written for it **is not** the logic you want. That's generating a valid URL for the purposes of running the spec via URL. The magic `integration` word is used there because in the URL Cypress knows which kind of spec file you're searching for on the file system.

However, for the purposes of what **you want** it is simply just `Cypress.spec.relative` which will always give you the relative path to file on the users hard drive relative from the `projectRoot`. You could chop off the `cypress/integration` part if you'd like using `path.resolve(config.integrationFolder, Cypress.spec.relative)`.

Reply:
@brian-mann Wow, really wish I knew about that earlier... completely simplifies a ton of stuff from this PR. I've now updated the PR to take advantage of `Cypress.spec.relative`.

As far as chopping off `cypress/integration` part, I think it's best to leave it now as it's consistent with what is shown in the error messages and code frame.

Reply:
agree